core/che-core-api-jdbc/src/main/java/org/eclipse/che/api/core/jdbc/jpa/guice/JpaInitializer.java

```diff
@@ -11,18 +11,30 @@
 package org.eclipse.che.api.core.jdbc.jpa.guice;
 
 import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import com.google.inject.name.Named;
 import com.google.inject.persist.PersistService;
 
+import java.nio.file.Paths;
+
 /**
  * Should be bound as eager singleton.
  * See <a href="https://github.com/google/guice/wiki/JPA">doc</a>
  *
  * @author Yevhenii Voevodin
  */
+@Singleton
 public class JpaInitializer {
 
+    @Inject(optional = true)
+    @Named("che.conf.storage")
+    private String storageRoot;
```

Comment:
as it's an optional injection, what will result for h2.baseDir if storageRoot is not defined ?